typing/includemod.ml

```diff
@@ -566,24 +566,25 @@ and functor_param ~in_eq ~loc env ~mark subst param1 param2 =
         | Ok (cc, _) -> Ok cc
         | Error err -> Error (Error.Mismatch err)
       in
-      let env, subst =
-        match name1, name2 with
-        | Some id1, Some id2 ->
-            Env.add_module id1 Mp_present arg2' env,
-            Subst.add_module id2 (Path.Pident id1) subst
-        | None, Some id2 ->
-            let id1 = Ident.rename id2 in
-            Env.add_module id1 Mp_present arg2' env,
-            Subst.add_module id2 (Path.Pident id1) subst
-        | Some id1, None ->
-            Env.add_module id1 Mp_present arg2' env, subst
-        | None, None ->
-            env, subst
-      in
+      let env, subst = equate_one_functor_param subst env arg2' name1 name2 in
```

Comment:
Nit: I would have `env` before `subst` in the parameter list, because `subst` lives "over" `env` (its well-formedness depends on the value of `env`).

Reply:
I would rather keep the same order as the one in the non-error path.